Refactor Key Vault Certificates based on review feedback

heaths opened the pull request:
Certificates portion for fixing #6078

heaths committed:
Refactor KeyVault Certificates from reviews

schaabs commented on sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateError.cs:
```diff
@@ -8,12 +8,19 @@ namespace Azure.Security.KeyVault.Certificates
     /// An error encountered during the processing of a <see cref="CertificateOperation"/>
     /// </summary>
-    public class Error : IJsonDeserializable
+    public class CertificateError : IJsonDeserializable
```
Possibly CertificateOperationError? I believe this only comes back as part of a CertificateOperationProperties

heaths replied:
That would be a change different from what we decided in the v-team before. We'll have to drive that across languages.

heaths replied:
Actually, given what we discussed earlier with @pakrym, shouldn't this just be captured in the updated `RequestFailedException`? Given the use case, it may not throw but could still be attached, much like in the TPL.

pakrym replied:
I think copying info from error to RequestFailedException would be great

schaabs commented on sdk/keyvault/Azure.Security.KeyVault.Certificates/tests/CertificateClientLiveTests.cs:
```diff
@@ -15,33 +14,12 @@ public CertificateClientLiveTests(bool isAsync) : base(isAsync)
-            certificatePolicy.IssuerName = "UNKNOWN";
-
-            await Client.StartCreateCertificateAsync(certName);
+            await Client.StartCreateCertificateAsync(certName, DefaultPolicy);
```
The default policy is not equivalent to the lines you reference above. DefaultPolicy has issuer type "Self", whereas this test relies on the certificate issuer being "UNKNOWN" so that the signing operation will not complete. Also the three tests below have the same issue.

heaths replied:
Thanks for catching. Likely just overzealous copy/paste.

heaths committed:
Resolve PR feedback and other pending changes

This doesn't include all the changes we made for Keys and Secrets GA, but close.

heaths committed:
Update changelog

heaths committed:
Revert changes to TestRecording
